src/Shared/DiagnosticSourceInstrumentation/PropertyFetcher.cs

```diff
@@ -105,10 +93,28 @@ static PropertyFetch CreateFetcherForProperty(PropertyInfo propertyInfo)
                     return null;
                 }
 
-                var typedPropertyFetcher = typeof(TypedPropertyFetch<,>);
-                var instantiatedTypedPropertyFetcher = typedPropertyFetcher.MakeGenericType(
-                    typeof(T), propertyInfo.DeclaringType, propertyInfo.PropertyType);
-                return (PropertyFetch)Activator.CreateInstance(instantiatedTypedPropertyFetcher, propertyInfo);
+                if (propertyInfo.DeclaringType!.IsValueType || propertyInfo.PropertyType.IsValueType || typeof(T).IsValueType)
```

Comment:
We should also add a check for:

```suggestion
                if (!RuntimeFeature.IsDynamicCodeSupported && (propertyInfo.DeclaringType!.IsValueType || propertyInfo.PropertyType.IsValueType || typeof(T).IsValueType))
```

This way we use the current approach when not on the NativeAOT runtime.